src/Contracts/Deployer/OpdexMarketDeployer.cs

```diff
@@ -30,57 +25,81 @@ public Address Owner
     /// <inheritdoc />
     public void SetOwner(Address address)
     {
-        Assert(Message.Sender == Owner, UnauthorizedError);
+        EnsureOwnerOnly();
         
         Owner = address;
         
         Log(new ChangeDeployerOwnerLog {From = Message.Sender, To = address});
     }
     
     /// <inheritdoc />
-    public Address CreateStandardMarket(Address marketOwner, bool authPoolCreators, bool authProviders, bool authTraders, uint fee)
+    public Address CreateStandardMarket(Address marketOwner, uint transactionFee, bool authPoolCreators, bool authProviders, bool authTraders, bool enableMarketFee)
     {
-        Assert(Message.Sender == Owner, UnauthorizedError);
-        
-        var response = Create<OpdexStandardMarket>(0, new object[] {marketOwner, authPoolCreators, authProviders, authTraders, fee});
-        
-        Assert(response.Success, InvalidMarketError);
+        EnsureOwnerOnly();
+
+        var marketParams = new object[] {transactionFee, marketOwner, authPoolCreators, authProviders, authTraders, enableMarketFee};
+        var marketResponse = Create<OpdexStandardMarket>(0, marketParams);
+
+        Assert(marketResponse.Success && marketResponse.NewContractAddress != Address.Zero, "OPDEX: INVALID_MARKET");
         
-        var market = response.NewContractAddress;
+        var market = marketResponse.NewContractAddress;
+        var router = CreateOpdexRouter(market, transactionFee, authProviders, authTraders);
 
         Log(new CreateMarketLog
         {
             Market = market, 
             Owner = marketOwner,
+            Router = router,
             AuthPoolCreators = authPoolCreators, 
             AuthProviders = authProviders, 
             AuthTraders = authTraders, 
-            Fee = fee,
-            StakingToken = Address.Zero
+            TransactionFee = transactionFee,
+            MarketFeeEnabled = enableMarketFee
         });
             
         return market;
     }
         
-    private void CreateStakingMarket(Address stakingToken)
+    /// <inheritdoc />
+    public Address CreateStakingMarket(Address stakingToken)
     {
+        EnsureOwnerOnly();
+        
         const uint transactionFee = 3; // .3% for the staking market
+        
+        Assert(State.IsContract(stakingToken), "OPDEX: INVALID_STAKING_TOKEN");
 
-        var response = Create<OpdexStakingMarket>(0, new object[] {stakingToken, transactionFee});
+        var marketParams = new object[] {transactionFee, stakingToken};
+        var marketResponse = Create<OpdexStakingMarket>(0, marketParams);
         
-        Assert(response.Success, InvalidMarketError);
+        Assert(marketResponse.Success && marketResponse.NewContractAddress != Address.Zero, "OPDEX: INVALID_MARKET");
```

Comment:
Surely if a `Create` call returns success then `NewContractAddress != Address.Zero` would _always_ be true.

Reply:
See this throughout the contracts

Reply:
This is resolved in both repos now.